src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/JS/WasmTestCommand.cs

```diff
@@ -67,33 +68,54 @@ protected override async Task<ExitCode> InvokeInternal(ILogger logger)
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
                 engineBinary = FindEngineInPath(engineBinary + ".cmd");
 
-            var engineArgs = new List<string>();
-
-            if (_arguments.Engine == JavaScriptEngine.V8)
+            var cts = new CancellationTokenSource();
+            try
             {
-                // v8 needs this flag to enable WASM support
-                engineArgs.Add("--expose_wasm");
-            }
+                ServerURLs? serverURLs = null;
+                if (_arguments.WebServerMiddlewarePathsAndTypes.Count > 0 || _arguments.SetWebServerEnvironmentVariablesHttp.Count > 0 || _arguments.SetWebServerEnvironmentVariablesHttps.Count > 0)
+                {
+                    serverURLs = await WebServer.Start(
+                        _arguments, logger,
+                        null,
+                        cts.Token);
+                    cts.CancelAfter(_arguments.Timeout);
+                }
 
-            engineArgs.AddRange(_arguments.EngineArgs);
-            engineArgs.Add(_arguments.JSFile);
+                var engineArgs = new List<string>();
 
-            if (_arguments.Engine == JavaScriptEngine.V8 || _arguments.Engine == JavaScriptEngine.JavaScriptCore)
-            {
-                // v8/jsc want arguments to the script separated by "--", others don't
-                engineArgs.Add("--");
-            }
+                if (_arguments.Engine == JavaScriptEngine.V8)
+                {
+                    // v8 needs this flag to enable WASM support
+                    engineArgs.Add("--expose_wasm");
+                }
```

Comment:
Why is this needed here? This should be added to the arguments in the runtime repo

Reply:
nvm, ignore this!

Reply:
it's not part of this change